python/taichi/main.py

```diff
@@ -72,6 +73,31 @@ def test_cpp(args):
     return int(task.run(*test_files))
 
 
+def run_example(name: list):
+    """Run an example based on the example NAME."""
+    import taichi as ti
+    from pathlib import Path
+
+    root_dir = ti.package_root() if ti.is_release() else ti.get_repo_directory(
+    )
+    examples_dir = Path(root_dir) / 'examples'
+    all_examples = examples_dir.rglob('*.py')
+    all_example_names = {
+        str(f.resolve()).split('/')[-1].split('.')[0]
+        for f in all_examples
+    }
+    if len(name) != 1:
+        sys.exit(
+            f"Sorry, please specify a name of example!\nAvailable examples names are: {all_example_names}"
+        )
+    elif name[0] not in all_example_names:
+        sys.exit(
+            f"Sorry, {name} is not an available example name!\nAvailable examples names are: {all_example_names}"
+        )
+    print(f"Running example {name[0]} ...")
+    importlib.import_module(f"examples.{name[0]}")
```

Comment:
Does this work if some example file has `if __name__ == '__main__'`?

Reply:
Great question, this, as well as other dynamic loading approaches, wouldn't work for the following examples:
{pbf2d.py, stable_fluid.py, particle_renderer.py, mpm_lagrangian_forces.py, mnist.py}
since they have either `main()` or `init()` wrapped with `if __name__ == '__main__'`.

Unfortunately, the easiest way I can think of is hard-coding this list and treat them conditionally and differently in the `run_example()` for now. I think I will go this way for this PR.

Ideally, if it's possible, maybe we could refactor the examples to let them all have `main()`, which will enable us to import them and call `main()` here? I was planning to make a separate issue for it.

Reply:
Actually I want to take back what I said above... I realized I should have used `runpy` for this type of tasks and don't need to hard-code anything 🤦 Let me test if it works, I'm just not sure if it will work in a release package.

Reply:
It seems `runpy` ends up being the saver here.